### DISCUSSION TITLE
Problem when modelling in 3D

### DISCUSSION THREAD

**Discussion by jacobosaa** [Anura3D/Anura3D_OpenSource#169] [General]
Hi everyone! I'm currently comparing two column collapses, one in 2D and the another one in 3D, using Anura3D_2023 with the exact same data, geometry, mesh size, material points, and everything runs pretty well for the 2D case, but when it comes to calculate the 3D model, this message appears. I've tried a lot of different approaches to this model, changing the mesh size or some other parameters but the error is the same. 
How can I fix it?

![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/7ef682b3-531b-4977-be1c-3554a70c4eaf)

Hoping to read from you soon!
Jacobo

**Comment by jacobosaa** [comment]
May I add that recently I've tried with an older version of Anura (2019) and this message appears

![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/128545000/cab9b604-e604-4eff-9c15-987b7062e89d)

**Comment by murphy-john** [comment]
Hi Jacobo! Have you tried using the debug feature in Visual Studio? You seem to be a very active user. Learning to debug really helped me figure out what, where, and why some of these errors were happening.

**Comment by jacobosaa** [comment]
Hi John! Well I'm very interested at this point about learning and
comprehending the software and the method, and right now I want to teach
this subject at my university.
Unfortunately,  I haven't used the debug tool mainly because I don't know
how. Let me put it this way: compiling the code to get the Anura3D 2023
versio. was a huge success!!
Do you know how can I learn to use thw debugging tool and how can I
implement it when a error occurs?
What do you think about teaching this as a subject? is it promising?

Thank you again for your help John.
Greetings from Chile!

**Reply by murphy-john** [reply]
Try starting [here.](https://github.com/Anura3D/Anura3D_OpenSource/discussions/29)

There are also many guides to be found searching on YouTube.

**Comment by enokjun** [comment]
I had the same problem when I first ran the ANURA3D with modeling done in GID.
Assuming the VS and fortran thing were properly installed, I managed to fix this issue by ensuring these steps:
1) the ANURA3D GiD_Problemtype folder was copied into the other Problemtypes in GID software
2) run ANURA3D Problemtypes before starting anything in GID software
3) open the calculate.bat file and ensure the folder path is accurate
Hope this helps!

**Comment by cvpontes** [comment]
Hello, @murphy-john 

How can we debug the Anura3D code? Where can we visualize/debug its running problems for the singular projects? In the .\src\VS\Anura3D.sln file? Instead of using the Release version inside the problemtype GiD folder, we should use a Debug build, so when we try to calculate, it shows up the line that's throwing the error, or something like that? What is the file we make the debugging process?

Thank you for your attention.

Carla.
